src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -274,13 +280,16 @@ static void uct_rdmacm_cm_handle_event_disconnected(struct rdma_cm_event *event)
     char               ip_port_str[UCS_SOCKADDR_STRING_LEN];
     char               ep_str[UCT_RDMACM_EP_STRING_LEN];
 
+    UCS_ASYNC_BLOCK(uct_rdmacm_cm_ep_get_async(cep));
     ucs_debug("%s: got disconnect event, status %d peer %s",
               uct_rdmacm_cm_ep_str(cep, ep_str, UCT_RDMACM_EP_STRING_LEN),
               event->status, ucs_sockaddr_str(remote_addr, ip_port_str,
                                              UCS_SOCKADDR_STRING_LEN));
 
     cep->disconnect_cb(&cep->super.super, cep->user_data);
-    cep->flags &= ~UCT_RDMACM_CM_EP_CONNECTED;
```

Comment:
why is this line removed?

Reply:
Otherwise no indication what connection is done, it's needed for flush.

Reply:
what do you mean?
it the disconnect cb was called, we can't say that the ep is still connected...

Reply:
Yes but we need indication that it was connected in the past, otherwise flush hangs

Reply:
restored